nphysics_testbed2d/src/testbed.rs

```diff
@@ -50,11 +52,11 @@ fn usage(exe_name: &str) {
 }
 
 pub struct Testbed {
-    world: World<f32>,
+    world: Arc<Mutex<World<f32>>>,
     window: Option<Box<Window>>,
     graphics: GraphicsManager,
     nsteps: usize,
-    callbacks: Vec<Box<Fn(&mut World<f32>, &mut GraphicsManager, f32)>>,
+    callbacks: Vec<Box<Fn(&mut GraphicsManager, f32)>>,
```

Comment:
I am not very fond of this, though I understand why this can be necessary to avoid dead-locks if, e.g., the user locks the physics world while inside of the callback. Maybe this could at least take `&mut Testbed` as its first argument, so the user can call `testbed.world()` to retrieve the shared world (see my first comment for the definition of `testbed.world()`).

Reply:
ok, or pass the `self.world` without locking it, to let the user choose between using the parameter or handling himself the dependency

Reply:
Yes, that will work too!

Reply:
For reference: passing ` &mut Testbed` in callback caused `cannot reborrow mutably` error and that's why I just plain removed it.

passing the `self.world` needs `Testbed` to be generic over its `WorldOwner` real type. I wanted to avoid adding complexity, but as you suggested it in other comments, I'll look into it.